src/components/CreateCourse/CreateCourse.jsx

```diff
@@ -115,93 +149,97 @@ const CreateCourse = ({
 		) {
 			alert('Please, fill in all fields');
 		} else {
-			addNewCourse((courseList) => [
-				...courseList,
-				{ ...newCourse, authors: courseAuthor.map((item) => item.id) },
-			]);
-			close(false);
+			newCourse = {
+				...newCourse,
+				duration: Number(duration),
+				authors: courseAuthor.map((item) => item.id),
+			};
+			fetchWithToken('courses/add', newCourse, token);
+			history.push('/courses');
```

Comment:
Please, go to courses page only after request that adds course is completed. Otherwise, it's possible that the user will see old data